pypesto/objective/objective.py

```diff
@@ -244,21 +264,34 @@ def _update_trace(self, x, mode, result):
         if self.trace is None:
             columns = ['time',
                        'n_fval', 'n_grad', 'n_hess', 'n_res', 'n_sres',
-                       'fval', 'grad', 'hess', 'res', 'sres',
-                       'x']
+                       'fval', 'grad', 'hess', 'res', 'sres', 'chi2', 'schi2',
+                       'x',]
             self.trace = pd.DataFrame(columns=columns)
 
         # extract function values
         if mode == Objective.MODE_FUN:
             fval = result.get(Objective.FVAL, None)
             grad = result.get(Objective.GRAD, None)
-            hess = None if self.options.trace_record_hess \
+            hess = None if not self.options.trace_record_hess \
                 else result.get(Objective.HESS, None)
             res = None
             sres = None
+            chi2 = None
+            schi2 = None
         else:  # mode == Objective.MODE_RES
-            res = result.get(Objective.RES, None)
-            sres = result.get(Objective.SRES, None)
+            res_result = result.get(Objective.RES, None)
+            sres_result = result.get(Objective.SRES, None)
+            res = None if not self.options.trace_record_res \
+                else res_result
+            sres = None if not self.options.trace_record_res \
+                else sres_result
+            chi2 = None if not self.options.trace_record_chi2 \
+                           or res_result is None \
+                else np.inner(res_result,res_result)/2
+            schi2 = None if not self.options.trace_record_chi2 \
+                            or sres_result is None \
+                            or res_result is None \
+                else res_result * res_result
```

Comment:
for chi2, can't we use the res_to_fval function? then the normalization is always the same. is the formula for schi2 correct? res_result * res_sresult maybe?